### PR TITLE
add kustomize for Iter8 install

### PR DESCRIPTION
**Pull request by kalantar**
## Description
Introduction of `kustomize` install scripts for [Iter8](https://iter8.tools) component.

## How Has This Been Tested?
Tested on RedHat OpenShift Local cluster. Manually created very basic kfdefs object and verified that Iter8 components were successfully created in the correct state.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: iter8/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Iter8 has a controller that automatically reconfigures the Service Mesh when the set of versions of a model change. This capability enables blue-green and canary rollouts of new versions of a model.
+
+## Installation process
+Following are the steps to install Iter8 as a part of OpenDataHub install:
+
+1. Install the OpenDataHub operator.
+2. Create a KfDef that includes the Iter8 component.
+
+```yaml
+apiVersion: kfdef.apps.kubeflow.org/v1
```
Comment (sriumcp) [inline iter8/README.md:26]: Taneem indicates that ODH v2.0 operator moved away from kfdef towards kustomize. Is this README still WIP? Will `kfdefs` eventually go away from this README to be replaced entirely by kustomize?

Comment (Jooho) [reply iter8/README.md:26]: yes, kfdef will go away so kustomize will be the main command to generate manifests.

**Comment by kalantar** [comment]
> @danielezonca Is this a planned feature for ODH Core or Model Mesh and included in a roadmap? Since we are bundling these manifests in the operator anything merged into this repo should have some documented alignment with ODH Core

The inclusion of Iter8 was reviewed and approved by the mlops sig; see https://github.com/opendatahub-io/opendatahub-community/pull/39. It is my understanding that this is as a tier2 project. If this is the wrong place for this, could you point me to the next steps? What is the appropriate approach to enabling a user to optionally deploy Iter8 as part of their ODH deployment?

**Comment by anishasthana** [comment]
Hey @kalantar , with the approval for tier 2, the right location to be would actually be https://github.com/opendatahub-io-contrib/odh-contrib-manifests. This is where tier 2 components are placed for incubation

**Comment by kalantar** [comment]
closing in favor of https://github.com/opendatahub-io-contrib/odh-contrib-manifests/pull/5 where I understand tier 2 community contributions should be made

**Comment by danielezonca** [comment]
Sorry I was on holiday, @anishasthana comment is correct, this repo is not intended for tier 2 component incubation.

@kalantar 
You are correct, with the new operator we are going to deprecate KfDef object and we plan to keep each manifest directly in the component repo (or in any case getting rid of the need of a centralised odh-manifests repo)
- https://github.com/opendatahub-io/opendatahub-operator/issues/338

But as you can see in the ticket it is still open there are still aspect to clarify, for example I don't know how the new operator is going to support tier2 incubation.

My suggestion is to continue with KfDef like you are doing even because most probably the manifests will be compatible (directly using kustomize) with the new operator so the migration should be smooth.

@VaishnaviHire @zdtsw @etirelli 
Do you have anything to add about Tier2 components and the new operator?

**Pull request closed without merging**